src/LegacyPortal.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import ReactDOM, { findDOMNode } from 'react-dom';
+import PropTypes from 'prop-types';
+
+const KEYCODES = {
+  ESCAPE: 27
+};
+
+const canUseDOM = !!(
+  typeof window !== 'undefined' &&
+  window.document &&
+  window.document.createElement
+);
+
+export default class Portal extends React.Component {
+  componentDidMount() {
+    this.renderPortal();
+  }
+
+  componentWillReceiveProps(props) {
+    this.renderPortal();
+  }
+
+  componentWillUnmount() {
+    if (this.props.node) {
+      document.body.removeChild(this.props.node);
+    }
+    if (this.defaultNode) {
+      document.body.removeChild(this.defaultNode);
+    }
+    this.defaultNode = null;
+    this.portal = null;
+  }
+
+  renderPortal(props) {
```

Comment:
Why not put this in render instead of calling it manually from componentDidMount and componentWillReceiveProps?

Reply:
calling it in render was giving warnings: `Warning: _renderNewRootComponent(): Render methods should be a pure function of props and state; triggering nested component updates from render is not allowed. If necessary, trigger nested updates in componentDidUpdate. Check the render method of Portal.`